src/bbox.rs

```diff
@@ -0,0 +1,117 @@
+use crate::vector::Vec3;
+
+const X_BIT: usize = 0b001usize;
+const Y_BIT: usize = 0b010usize;
+const Z_BIT: usize = 0b100usize;
+
+pub struct BBox {
+    left: f32,
```

Comment:
Why am I not using Vec3s here? `min_dimensions` and `max_dimensions` instead of 6 individual values.